### ISSUE TITLE
Diagnostics 'check arguments to R function calls' missed piped argument in some cases

### ISSUE THREAD

**Issue by MarekGierlinski** [rstudio/rstudio#11780]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.07.1 build 554
    OS Version      : MacOS version 10.15.7
    R Version       : 4.2

### Steps to reproduce the problem

1. Install packages `dendsort`, `ggdendro` and `tibble`. The error only appears when packages are installed.

2. Create the following code in Rstudio and save it (the error only appears after saving). Also, it needs an empty line at the end.

```

dendr <- dat |> 
  dendsort::dendsort() |> 
  ggdendro::dendro_data()

lst <- dat |>
  tibble::as_tibble() |> 
  tibble::deframe()

```

3. Enable Tools -> Global options -> Code -> Diagnostics -> Check arguments to R function calls

The yellow warning triangles will appear next to `dendsort` and `deframe`. The warning is about the missing argument of the function. It only appears with these functions, not with others.

![](https://github.com/MarekGierlinski/images/blob/main/Screenshot%202022-08-19%20at%2011.46.26.png?raw=true)

### Describe the problem in detail

Normally, diagnostics recognises function arguments passed via R pipe (with `|>` or `%>%`). However, with it fails with `dendsort` and `deframe`. There might be other functions too, but this is what I discovered. For these two functions it generates a warning.

### Describe the behavior you expected

As with other functions, diagnostics should recognise piped argument and not raise a warning.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by MarekGierlinski** [comment]
Just discovered that `str_remove` cases the same warning.

![](https://github.com/MarekGierlinski/images/blob/main/Screenshot%202022-08-19%20at%2011.59.14%20(2).png?raw=true)

**Comment by ronblum** [comment]
@MarekGierlinski Thank you for raising this! I can reproduce this in RStudio Desktop Spotted Wakerobin 2022.07.2-576 on MacOS 13, as well. However, it's working in build 2022.11.0-daily-169 for the Elsbeth Geranium release currently in development. I you're interested in trying it out, [daily builds are available for it](https://dailies.rstudio.com/rstudio/elsbeth-geranium/electron/macos/). However, please note that these are in active development and meant for testing.

**Comment by ronblum** [comment]
This was fixed by @kevinushey , not a new issue that magically went away. So in that case:

Confirmed that it's fixed in RStudio Desktop 2022.11.0-daily-169 on MacOS 13.
Verified using the code above. (I added a value for `dat` to make the "no symbol named 'dat' in scope" messages go away.)

<img width="236" alt="dat" src="https://user-images.githubusercontent.com/31009952/189433172-2228fe01-7d67-45a9-9dec-fb7d35466dcb.png">
